basic.go

```diff
@@ -21,15 +32,486 @@ func (cb *Cinnabot) Echo(msg *message) {
+			"/nusbus: nus bus timings for bus stops around your location\n" +
+			"/weather: 2h weather forecast\n" +
+			"/resources: list of important resources!\n" +
+			"/spaces: list of space bookings\n" +
+			"/feedback: to give feedback\n\n" +
+			"_*My creator actually snuck in a few more functions🕺 *_\n" +
+			"Try using /help <func name> to see what I can _really_ do"
+	cb.SendTextMessage(int(msg.Chat.ID), text)
 }
 
 // About returns a link to Cinnabot's source code.
 func (cb *Cinnabot) About(msg *message) {
-	cb.SendTextMessage(msg.From.ID, "Touch me: https://github.com/varunpatro/Cinnabot")
+	cb.SendTextMessage(int(msg.Chat.ID), "Touch me: https://github.com/varunpatro/Cinnabot")
 }
 
-// Capitalize returns a capitalized form of the input string.
-func (cb *Cinnabot) Capitalize(msg *message) {
-	cb.SendTextMessage(msg.From.ID, strings.ToUpper(strings.Join(msg.Args, " ")))
+//Link returns useful resources
+func (cb *Cinnabot) Resources(msg *message) {
+	resources := make(map[string]string)
+	resources["usplife"] = "[fb page](https://www.facebook.com/groups/usplife/)"
+	resources["food"] = "@rcmealbot"
+	resources["spaces"] = "[spaces web](http://www.nususc.com/Spaces.aspx)"
+	resources["usc"] = "[usc web](http://www.nususc.com/MainPage.aspx)"
+	resources["study groups"] = "@uyp\\_bot"
+
+	var key string = strings.ToLower(strings.Join(msg.Args, " "))
+	log.Print(key)
+	_, ok := resources[key]
+	if ok {
+		cb.SendTextMessage(int(msg.Chat.ID), resources[key])
+	} else {
+		var values string = ""
+		for key, _ := range resources {
+			values += key + " : " + resources[key] + "\n"
+		}
+		msg := tgbotapi.NewMessage(msg.Chat.ID, values)
+		msg.DisableWebPagePreview = true
+		msg.ParseMode = "markdown"
+		cb.SendMessage(msg)
+	}
+}
+
+//Structs for weather forecast function
+type WeatherForecast struct {
+	AM []AreaMetadata `json:"area_metadata"`
+	FD []ForecastData `json:"items"`
+}
+
+type AreaMetadata struct {
+	Name string            `json:"name"`
+	Loc  tgbotapi.Location `json:"label_location"`
+}
+
+type ForecastData struct {
+	FMD []ForecastMetadata `json:"forecasts"`
+}
+
+type ForecastMetadata struct {
+	Name     string `json:"area"`
+	Forecast string `json:"forecast"`
+}
+
+//Weather checks the weather based on given location
+func (cb *Cinnabot) Weather(msg *message) {
+	//Check if weather was sent with location, if not reply with markup
+	if len(msg.Args) == 0 || !cb.CheckArgCmdPair("/weather", msg.Args) {
+		opt1 := tgbotapi.NewKeyboardButtonRow(tgbotapi.NewKeyboardButton("Cinnamon"))
+		opt2B := tgbotapi.NewKeyboardButton("Here")
+		opt2B.RequestLocation = true
+		opt2 := tgbotapi.NewKeyboardButtonRow(opt2B)
+
+		options := tgbotapi.NewReplyKeyboard(opt1, opt2)
+
+		replyMsg := tgbotapi.NewMessage(int64(msg.Message.From.ID), "🤖: Where are you?\n\n")
+		replyMsg.ReplyMarkup = options
+		cb.SendMessage(replyMsg)
+		return
+	}
+
+	//Default loc: Cinnamon
+	loc := &tgbotapi.Location{Latitude: 1.306671, Longitude: 103.773556}
```

Comment:
`const`

Reply:
Constant can only be declared on character, string, boolean values, so I can only make the latitude and longitude a constant value.
Since this is the only only use of it in this file, where should I declare the constant? And how should I do that.

Reply:
declare two `const`s at the top of the file:

```go
const (
    cinnamonLat := 1.306671
    cinnamonLong := 103.773556
)
```